misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -127,45 +124,47 @@ public ModuleCoverage generateReport(JarResolver jarResolver) throws IOException
         }
     }
 
-    private IBundleCoverage analyzeStructure() throws IOException {
+    private CoverageBuilder analyzeStructure() throws IOException {
         final CoverageBuilder coverageBuilder = new CoverageBuilder();
         final Analyzer analyzer = new Analyzer(execFileLoader.getExecutionDataStore(), coverageBuilder);
         analyzer.analyzeAll(classesDirectory.toFile());
-        return coverageBuilder.getBundle(title);
+        return coverageBuilder;
     }
 
-    private void createXMLReport(IBundleCoverage bundleCoverage) throws IOException {
-            XMLFormatter xmlFormatter = new XMLFormatter();
-            File reportFile = new File(target.getReportPath().resolve(
-                    this.module.moduleName().toString()).resolve(REPORT_XML_FILE).toString());
-            reportFile.getParentFile().mkdirs();
-
-            try (FileOutputStream fileOutputStream = new FileOutputStream(reportFile)) {
-                IReportVisitor visitor = xmlFormatter.createVisitor(fileOutputStream);
-                visitor.visitInfo(execFileLoader.getSessionInfoStore().getInfos(),
-                        execFileLoader.getExecutionDataStore().getContents());
-
-                visitor.visitBundle(bundleCoverage, null);
+    private IBundleCoverage getPartialCoverageModifiedBundle(CoverageBuilder coverageBuilder) {
+        final Collection<ISourceFileCoverage> sourcefiles = coverageBuilder.getSourceFiles();
+        final Collection<ISourceFileCoverage> modifiedSourceFiles = modifySourceFiles(sourcefiles);
+        final IBundleCoverage bundleCoverage = new BundleCoverageImpl(title, coverageBuilder.getClasses(),
+                modifiedSourceFiles);
+        return bundleCoverage;
```

Comment:
shall we remove redundant local variables. we can write an inline return here.
```java
return new BundleCoverageImpl(title, coverageBuilder.getClasses(),
                modifySourceFiles(coverageBuilder.getSourceFiles()));
```

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/28767/commits/b33977232ac9c95a1b70e8307c3c9eb5133bde89